### DISCUSSION TITLE
1.21 leaving default paintings and adding extra with new components system

### DISCUSSION THREAD

**Discussion by khaim0919** [jthistle/mc-paintings#105] [General]
Is there a plan to implement the components system into this website to allow for the vanilla paintings and custom ones in the same place? I am interesting in helping with that and the custom music discs now that 1.21 is out. 

The ability to create a resource pack and data pack with both the paintings and music discs (adding sounds and music disc textures) would be a godsend of a site. 

This [safeliquid's custom music discs ](https://github.com/safeliquids/custom-music-discs) provides an example of how the components system can be used to create custom discs and [TeamTernate's infinite-music-discs](https://github.com/TeamTernate/infinite-music-discs) has an good UI and generates both a resource pack and datapack though for 1.20.4 and under.

**Comment by jthistle** [comment]
There's an issue with the way paintings work: although there is technically the ability to have infinite paintings, the way that players place paintings is by using the 'painting' item, which picks a random painting from the set of built-in ones _only_. So if I were to add new paintings using the variants system, these would not be accessible in survival, and in creative only with console commands (since the creative menu cannot be modified). Of course, power users will be alright with this, but for the vast majority who play Minecraft as-is, this is not acceptable. For this reason, we have to stick to the system of overwriting vanilla painting textures.

The trouble then for me is: how do we make the website work for both cases, for people who want to be able to play in survival without commands, and for people who want to utilise the full system available? I'm interested in your thoughts.

As for music discs, this is a whole different can of worms (and thanks for the links to these repos, they look useful). I need to have a good think about how I might implement this in tandem with the paintings part of the website, but I'm confident it shouldn't be too hard. I'll prototype a few things when I get the time, but after I've got basic functionality working you're more than welcome to contribute code.
